pennylane/gradients/pulse_gradient.py

```diff
@@ -162,8 +162,8 @@ def _diff_and_contract(res_list, cjacs, int_prefactor):
             diff = (res := qml.math.stack(res_list))[::2] - res[1::2]
             return qml.math.tensordot(diff, cjacs, axes=[[0], [0]]) * int_prefactor
 
-    # If multiple measure xor shot_vector: One axis to pull out of the shift rule and integration
-    if not single_measure + shot_vector == 1:
+    # If single measure xor has_partitioned_shots: One axis to pull out of the shift rule and integration
+    if single_measure is has_partitioned_shots:
```

Comment:
Trying to figure out is we can make this code a little more straightforward and readable.

What we care about is "how many layers of nesting is this feature adding?" `multiple_measure = not single_measure` adds a layer of nesting. `has_partitioned_shots` adds an additional layer of nesting.

I wonder if we could create some explanatory variables:
```
multiple_measure = not single_measure
layers_of_nesting = multiple_measure + has_partitioned_shots
```

Then below we can do dispatch on:
```
if layers_of_nesting == 1:
     # first block
if layers_of_nesting == 0:
    # second block
# final block = layers_of_nesting == 2
```

This is probably outside the scope of this PR, but I just wanted to understand what this code was actually doing.

Reply:
i've updated this PR to match what you've said - I'll take self-explanatory code over a comment any day of the week